cmake/kokkos_tpls.cmake

```diff
@@ -28,41 +30,26 @@ SET(HPX_DEFAULT OFF)
 ENDIF()
 KOKKOS_TPL_OPTION(HPX ${HPX_DEFAULT})
 
-IF (KOKKOS_ENABLE_HPX)
-  FIND_PACKAGE(HPX REQUIRED)
-#  MESSAGE(STATUS "KOKKOS_ENABLE_HPX: ${HPX_DIR}")
-ENDIF()
-
-IF (KOKKOS_ENABLE_HWLOC)
-  FIND_PACKAGE(HWLOC REQUIRED MODULE)
-  IF (NOT HWLOC_FOUND)
-    MESSAGE(FATAL_ERROR "Unable to locate hwloc ${KOKKOS_HWLOC_DIR}")
-  ENDIF()
-ENDIF()
-
-IF (KOKKOS_ENABLE_LIBNUMA)
-  FIND_PACKAGE(LIBNUMA REQUIRED MODULE)
-  IF (NOT LIBNUMA_FOUND)
-    MESSAGE(FATAL_ERROR "Unable to locate libnuma ${KOKKOS_LIBNUMA_DIR}")
-  ENDIF()
-ENDIF()
-
-IF (KOKKOS_ENABLE_LIBRT)
-  FIND_PACKAGE(LIBRT REQUIRED MODULE)
-  IF (NOT LIBRT_FOUND)
-    MESSAGE(FATAL_ERROR "Unable to locate LIBRT ${KOKKOS_LIBRT_DIR}")
-  ENDIF()
-ENDIF()
-
-FIND_PACKAGE(LIBDL REQUIRED MODULE)
-IF (NOT LIBDL_FOUND)
-  MESSAGE(FATAL_ERROR "Unable to locate ldl ${KOKKOS_LIBDL_DIR}")
-ENDIF()
-
-IF (KOKKOS_ENABLE_MEMKIND)
-  FIND_PACKAGE(MEMKIND REQUIRED MODULE)
-  IF (NOT MEMKIND_FOUND)
-    MESSAGE(FATAL_ERROR "Unable to locate memkind ${KOKKOS_MEMKIND_DIR}")
-  ENDIF()
-ENDIF()
-
+IF(Trilinos_ENABLE_Kokkos AND TPL_ENABLE_PTHREAD)
+SET(PTHREAD_DEFAULT ON)
+ELSE()
+SET(PTHREAD_DEFAULT OFF)
+ENDIF()
+KOKKOS_TPL_OPTION(PTHREAD ${PTHREAD_DEFAULT})
+
+
+#Make sure we use our local FindKokkosCuda.cmake
+KOKKOS_IMPORT_TPL(HPX INTERFACE)
+KOKKOS_IMPORT_TPL(CUDA INTERFACE)
+KOKKOS_IMPORT_TPL(HWLOC)
+KOKKOS_IMPORT_TPL(LIBNUMA)
+KOKKOS_IMPORT_TPL(LIBRT)
+KOKKOS_IMPORT_TPL(LIBDL)
+KOKKOS_IMPORT_TPL(MEMKIND)
+KOKKOS_IMPORT_TPL(PTHREAD INTERFACE)
+
+#Convert list to newlines (which CMake doesn't always like in cache varibles)
+STRING(REPLACE ";" "\n" KOKKOS_TPL_EXPORT_TEMP "${KOKKOS_TPL_EXPORTS}")
+#Convert to a regular variable
```

Comment:
Why would `KOKKOS_TPL_EXPORTS` be in the cache at all?  Is that because of the `GLOBAL_SET` function?  Did you consider not using it?

Reply:
Yes, feedback was that we should probably append to a global variable due to function scope issues down the line.  I did consider doing this as just a regular variable (along with a bunch of others).  It seem liked using an `INTERNAL` cache variable with global scope would be less error prone.  

CMake doesn't like newlines in cache variables - so for the `@` replacement it has to be a regular variable.